tools/azsdk-cli/docs/specs/0-core-telemetry.spec.md

```diff
@@ -0,0 +1,1543 @@
+```
+
+**Agent Activity:**
+1. Agent calls `azsdk_generate_sdk` for Python storage
+2. Agent calls `azsdk_run_tests` in playback mode
+3. Agent calls `azsdk_run_pr_checks` to validate
+4. Multiple telemetry events generated (one per tool call)
+
+**Telemetry Events Generated:**
+```json
+// Event 1: SDK Generation
+{
+  "timestamp": "2025-10-16T15:00:00.000Z",
+  "name": "ToolExecuted",
+  "duration": 10500,
+  "customDimensions": {
+    "sessionId": "xyz123",
+    "toolName": "azsdk_generate_sdk",
+    "language": "Python",
+    "serviceName": "storage"
+  }
+}
+
+// Event 2: Test Execution
+{
+  "timestamp": "2025-10-16T15:00:11.500Z",
+  "name": "ToolExecuted",
+  "duration": 6200,
+  "customDimensions": {
+    "sessionId": "xyz123",
+    "toolName": "azsdk_run_tests",
+    "language": "Python",
+    "serviceName": "storage"
+  }
+}
+
+// Event 3: PR Checks
+{
+  "timestamp": "2025-10-16T15:00:18.700Z",
+  "name": "ToolExecuted",
+  "duration": 15000,
+  "customDimensions": {
+    "sessionId": "xyz123",
+    "toolName": "azsdk_run_pr_checks",
+    "language": "Python",
+    "serviceName": "storage"
+  }
+}
+```
+
+**Key Telemetry Insights:**
+- Complete workflow adoption (how many users do full workflow vs. partial?)
+- Tool call sequences (what order do users execute tools?)
+- Where users drop off in the workflow
+- Time to complete full workflow
+- Success rate of end-to-end workflow
+
+### Prompt 5: Update Package Metadata
+
+**User Prompt:**
+```text
+Update the package version to 2.0.0-beta.1 for the Cosmos DB management plane SDK in all languages.
+```
+
+**Agent Activity:**
+1. Agent calls `azsdk_update_package` tool
+2. Tool updates version in metadata files
+3. Tool updates changelogs (mgmt plane only)
+
+**Telemetry Event Generated:**
+```json
+{
+  "timestamp": "2025-10-16T16:30:22.890Z",
+  "name": "ToolExecuted",
+  "success": true,
+  "duration": 2100,
+  "customDimensions": {
+    "toolName": "azsdk_update_package",
+    "toolCategory": "package",
+    "toolArgs": "{\"serviceName\":\"cosmosdb\",\"version\":\"2.0.0-beta.1\",\"languages\":\"all\"}",
+    "planeType": "mgmt",
+    "serviceName": "cosmosdb"
+  }
+}
+```
+
+**Key Telemetry Insights:**
+- Package update frequency by service
+- Preview vs. GA release ratio
+- Which services release most frequently
+
+---
+
+## CLI Commands
+
+This section demonstrates direct CLI usage and the telemetry events generated. CLI telemetry uses the same schema as MCP server telemetry but sets `platform: "CLI"`.
+
+### Command 1: Verify Setup
+
+**CLI Command:**
+```bash
+azsdk verify-setup --languages .NET,Java,JavaScript,Python,Go --verbose
+```
+
+**Telemetry Event Generated:**
+```json
+{
+  "timestamp": "2025-10-16T14:20:45.678Z",
+  "name": "ToolExecuted",
+  "success": true,
+  "duration": 3210,
+  "customDimensions": {
+    "platform": "CLI",
+    "toolName": "azsdk_verify_setup",
+    "toolCategory": "env-setup",
```

Comment:
I wonder if we want to support tool annotations in source for this, or base category on some sort of CLI hierarchy.